### PR TITLE
Update HttpMethod to include all known HTTP verbs listed on RFCs

### PR DESCRIPTION
**Pull request by pmlopes**
Currently `HttpMethod` includes the verbs defined in 2 RFCs:

* RFC 2616
* RFC 5789

This pull request adds a few more scattered on other RFCs

* RFC 2518
* RFC 4918
* RFC 4791
* RFC 3253
* RFC 3648
* RFC 3744
* RFC 5323

This allows users to build more than "traditional" web servers with vertx, but for example implement `dav` servers, etc...

### CODE REVIEW

**Comment by Stwissel** [comment]
I highly appreciate the additional keyword support. It allows us to implement http based round-trip file editing (webDAV - the same approach Sharepoint takes) and support for calDAV to have read/write support for calendar entries.

File: src/main/java/io/vertx/core/http/HttpMethod.java [hunk 1]
```diff
@@ -13,12 +13,98 @@
...
+
+  private static final Map<String, HttpMethod> METHOD_MAP = new HashMap<>();
+
+  static {
+    for (HttpMethod method : HttpMethod.values()) {
+      METHOD_MAP.put(method.toString(), method);
+    }
+  }
+
+  public static HttpMethod fromHTTPProtocol(String name) {
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/HttpMethod.java]: missing javadoc

**Comment by vietj** [comment]
I'm wondering if instead we should not make this enum a plain Java interface instead

**Comment by vietj** [comment]
code would not break except when people used it as a switch in an enum, that they could replace with a String switch. That would allow getting rid of HttpMethod.OTHER and allow to plug any HTTP method.

**Comment by pmlopes** [comment]
See #3252

**Pull request closed without merging**